corehq/motech/repeaters/dbaccessors.py

```diff
@@ -90,6 +115,19 @@ def get_paged_repeat_records(domain, skip, limit, repeater_id=None, state=None):
     return [RepeatRecord.wrap(result['doc']) for result in results]
 
 
+def get_paged_sql_repeat_records(domain, skip, limit, repeater_id=None, state=None):
+    from .models import SQLRepeatRecord
+
+    queryset = SQLRepeatRecord.objects.filter(domain=domain)
+    if repeater_id:
+        queryset = queryset.filter(repeater_stub__repeater_id=repeater_id)
+    if state:
+        queryset = queryset.filter(state=state)
+    return (queryset.order_by('-registered_at')[skip:skip + limit]
```

Comment:
I share your concern.

Thanks for the links. I also looked at [Ethan's recent PR](https://github.com/dimagi/commcare-hq/pull/29396) along the same lines. But all of them seem to me to be appropriate for iterating rows or models, but not for paginating a report.

I think we have two options here:
* Use Elasticsearch instead of Postgres for the Repeat Record Report.
* Hope that users don't care about what happens more than a couple of pages into the report.

I don't really love either of those options. Are my concerns valid? Are there other options I'm not considering? 

@dannyroberts @gherceg do you have opinions on whether the Repeat Records Report should be backed by Elasticsearch or Postgres? And if Postgres, how we should paginate it?

Reply:
I wonder, doesn't couch (which currently backs the report) have the same pagination issue? And doesn't elasticsearch? Maybe I'm missing some special magic that Couch and/or Elasticsearch do, but my guess is that they also just internally do the query for `skip + limit` items and return the last `limit` of them.

At some point it also becomes more than is feasible for a person to actually click through. So having more granular filters and an upper limit (10k? 100k?) to the number of items we're willing to show in the results through pagination seems like the best longer term option to me